apisix/init.lua

```diff
@@ -391,6 +400,17 @@ local function verify_https_client(ctx)
                            ", but the host is ", host)
             return false
         end
+
+        local session_hostname, err = apisix_ssl.session_hostname()
+        if err then
+            core.log.error("failed to get session hostname: ", err)
+            return false
+        end
+        if session_hostname and session_hostname ~= ngx.ctx.client_hello_sni then
+            core.log.error("sni in client hello mismatch hostname of ssl session, ",
+                             "sni: ", ngx.ctx.client_hello_sni, ", hostname: ", session_hostname)
+            return false
+        end
```

Comment:
What is the difference between this and the other modification that looks the same above?

Reply:
The above is the http subsystem, and this is the stream subsystem.

Reply:
The HTTP subsystem also has matching verification for the HTTP host and SNI, so the verification logic of these two subsystems is not exactly the same and cannot be fully reused.

Reply:
> Maybe they could be moved to a new function with some comments explaining why this verification is done "to fix a security issue".

done